### PR TITLE
Add Filter string generation

### PR DESCRIPTION
**Pull request by zdypro888**
### General:

* [ ] Have you removed all sensitive information, including but not limited to access keys and passwords?
* [ ] Have you checked to ensure there aren't other open or closed [Pull Requests](../../pulls) for the same bug/feature/question?

### Pull Request Guidelines:

1. [ ] Does your submission pass tests?
1. [ ] Have you added tests? 
1. [ ] Are you addressing a single feature in this PR? 
1. [ ] Are your commits atomic, addressing one change per commit?
1. [ ] Are you following the conventions of the language? 
1. [ ] Have you saved your large formatting changes for a different PR, so we can focus on your work?
1. [ ] Have you explained your rationale for why this feature is needed? 
1. [ ] Have you linked your PR to an [open issue](https://blog.github.com/2013-05-14-closing-issues-via-pull-requests/)

### CODE REVIEW

**Comment by phillc** [comment]
Hello,
This is an interesting idea. We are circulating this internally for feedback. Thank you for the PR!

File: comparison.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+		return fmt.Sprintf("{\"%s\": {\"%s\": \"%s\"}",
+			c.Column, c.Operator, c.Value)
+	}
+	if _, ok := c.Value.(int); ok {
+		return fmt.Sprintf("{\"%s\": {\"%s\": %d}",
+			c.Column, c.Operator, c.Value)
+	}
+
+	return ""
+}
```
Comment (LBGarber) [inline comparison.go:118]:
```suggestion
func (c *Comparison) JSON() string {
	if c.Operator == Eq {
		return fmt.Sprintf("{\"%s\": %s}", c.Column, getJSONValueString(c.Value))
	}

	return fmt.Sprintf("{\"%s\": {\"%s\": %v}",
		c.Column, c.Operator, getJSONValueString(c.Value))
}

File: comparison.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+func And(nodes ...FilterNode) *Filter {
+	return &Filter{LogicalAnd, nodes}
+}
+
+func Or(nodes ...FilterNode) *Filter {
+	return &Filter{LogicalOr, nodes}
+}
```
Comment (LBGarber) [inline comparison.go:126]:
```suggestion
}

func getJSONValueString(value interface{}) string {
	if _, ok := value.(string); ok {
		return fmt.Sprintf("\"%s\"", value)
	}

	return fmt.Sprintf("%v", value)
}
```

**Review by LBGarber — CHANGES_REQUESTED** [review]
I love this idea! Here are a few ways we can simplify the logic.

**Review by LBGarber — APPROVED** [review]
Looks good!!

**Pull request merged**